app/templates/base.html

```diff
@@ -37,16 +37,16 @@
             <ul class="list">
                 {% if current_user.is_authenticated and current_user.urole == 'student' %}
                 <li class="navitem"><a href="{{ url_for('dashboard') }}" class="link">Home</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#viewclass" class="link">View Classes</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#joinclass" class="link">Join A Class</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#topicalquiz" class="link">Topical Quizzes</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#mockquiz" class="link">Mock Quizzes</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='viewclass') }}" class="link">View Classes</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='joinclass') }}" class="link">Join A Class</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='topicalquiz') }}" class="link">Topical Quizzes</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='mockquiz') }}" class="link">Mock Quizzes</a></li>
                 <li class="navitem"><a href="{{ url_for('logout') }}" class="link">Sign Out</a></li>
                 {% elif current_user.is_authenticated and current_user.urole == 'educator' %}
                 <li class="navitem"><a href="{{ url_for('dashboard') }}" class="link">Home</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#viewclasses" class="link">View Classes</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#createclass" class="link">Create A Class</a></li>
-                <li class="navitem"><a href="{{ url_for('dashboard') }}#createquestion" class="link">Create Questions</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='viewclass') }}" class="link">View Classes</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='createclass') }}" class="link">Create A Class</a></li>
+                <li class="navitem"><a href="{{ url_for('dashboard', _anchor='createquestion') }}" class="link">Create Questions</a></li>
                 <li class="navitem"><a href="{{ url_for('logout') }}" class="link">Sign Out</a></li>
                 {% else %}
                 <li class="navitem"><a href="{{ url_for('home') }}" class="link">Home</a></li>
```

Comment:
can we abstract it?

`{{% if current_user.is_authenticated %}}`
home
view classes
    `{{% if current_user.urole == 'student' %}}`
    ...
    `{{% elif current_user.urole == 'educator' %}}`
    ...